### ISSUE TITLE
CVE-2020-8558

### ISSUE THREAD

**Issue by travis-sobeck** [rancher/rancher#27875]
Just a quick question about this [CVE](https://github.com/kubernetes/kubernetes/issues/92315). Will this be addressed in v2.4.6?  In the meantime, what's the best approach to mitigation if running v2.4.4 or v2.4.5?
Thanks

**Comment by maggieliu** [comment]
@travis-sobeck We are currently investigating the impact of the CVE on Rancher. Should have a mitigation plan / path forward soon.

**Comment by travis-sobeck** [comment]
Sounds good, thanks.

**Comment by maggieliu** [comment]
We are working to update Rancher to address the following CVE:

- [CVE - 2020-8558](https://github.com/kubernetes/kubernetes/issues/92315): Node setting allows for neighboring hosts to bypass localhost boundary

CVE-2020-8558 was patched in the following versions of Kubernetes: v1.18.4+, v1.17.7+, v1.16.11+

To ensure you have the latest stable version of Kubernetes, for Rancher v2.4.5, the following k8s versions will be made available using KDM: 
- v1.18.5-rancher1-1
- v1.17.8-rancher1-1
- v1.16.12-rancher1-1 

For Rancher v2.3.8, the following k8s versions will be made available using KDM
- v1.17.8-rancher1-1
- v1.16.12-rancher1-1

**Comment by travis-sobeck** [comment]
KDM?

**Comment by maggieliu** [comment]
@travis-sobeck it's updating our Kontainer-driver-metadata, which will automatically update and installed 2.4.5 and 2.3.8 Rancher version. However we are currently running into a technically issue with the latest patches. So there might be a delay in delivering this.

**Comment by maggieliu** [comment]
Blocker issue with the latest k8s patches: https://github.com/rancher/rancher/issues/27652

**Comment by travis-sobeck** [comment]
Ah, ok.  I thought maybe it was some CLI tool that I need to run against my cluster instead of helm upgrade.

**Comment by superseb** [comment]
Currently we are still investigating a blocker issue caused by the latest patches. In the meantime, users can use the following information to mitigate the vulnerability. See the original issue created by the Kubernetes team here: https://github.com/kubernetes/kubernetes/issues/92315

### Am I vulnerable?

- You are running a vulnerable version
  - Kubernetes v1.18.0 - v1.18.3
  - Kubernetes v1.17.0 - v1.17.6
  - Kubernetes <= v1.16.10
- Your cluster nodes run in an environment where untrusted hosts share the same layer 2 domain (i.e. same LAN) as nodes
  - To be decided by the user
- Your cluster allows untrusted pods to run containers with `CAP_NET_RAW` (the Kubernetes default is to allow this capability).
  - This is not possible when using the `restricted` PodSecurityPolicy
  - If you are not using the `restricted` PodSecurityPolicy, you can disallow the use of this capability

See https://rancher.com/docs/rancher/v2.x/en/admin-settings/pod-security-policies/ for more information about configuring PodSecurityPolicy

- Your nodes (or hostnetwork pods) run any localhost-only services which do not require any further authentication. To list services that are potentially affected, run the following commands on nodes:

  - `lsof +c 15 -P -n -i4TCP@127.0.0.1 -sTCP:LISTEN`
  - `lsof +c 15 -P -n -i4UDP@127.0.0.1`

  On a master node, an lsof entry like this indicates that the API server may be listening with an insecure port:

```
COMMAND        PID  USER FD   TYPE DEVICE SIZE/OFF NODE NAME
kube-apiserver 123  root  7u  IPv4  26799      0t0  TCP 127.0.0.1:8080 (LISTEN)
```

### How do I mitigate this vulnerability?

Prior to upgrading, this vulnerability can be mitigated by manually adding an iptables rule on nodes. This rule will reject traffic to 127.0.0.1 which does not originate on the node.

```
 iptables -I INPUT --dst 127.0.0.0/8 ! --src 127.0.0.0/8 -m conntrack ! --ctstate RELATED,ESTABLISHED,DNAT -j DROP
```

Additionally, if your cluster does not already have the API Server insecure port disabled, we strongly suggest that you disable it.

The configuration parameter for kube-apiserver to disable the insecure port is `--insecure-port=0` . This is the default for all currently released Kubernetes versions in v2.4.5. This can be verified using the following command (requires `jq` to be installed):

```
docker inspect kube-apiserver | jq -e '.[0].Args[] | match("--insecure-port=0").string'
```

This should result in the output, confirming the configuration parameter is present:

```
"--insecure-port=0"
```

If you need to manually apply this configuration, please follow the documentation as described on https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/rke-clusters/options/#cluster-config-file using the following configuration:

```
services:
  kube-api:
    insecure-port: "0"
```

###

**Comment by maggieliu** [comment]
Tracking all pending CVEs here: https://github.com/rancher/rancher/issues/27950
